### PR TITLE
C3: Adapt Astro c3 flow to their v10 Cloudflare adapter

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
## What this PR solves / how to test

This PR updates c3 to handle the upcoming Astro v10 cloudflare adapter (https://github.com/withastro/adapters/pull/159)

To test the changes in this PR:
 - create a new Astro app using C3 (opting in to typescript to also test the types update)
    ```
    npx https://prerelease-registry.devprod.cloudflare.dev/workers-sdk/runs/8452603565/npm-package-create-cloudflare-5410 --no-auto-update
    ```
 - update the `@astrojs/cloudflare` version to `astrojs/cloudflare@experimental--cf-10`:
   ```
   npm i -D @astrojs/cloudflare@experimental--cf-10
   ```
 - make sure that everything works as expected

> [!NOTE]
> The C3 e2es are currently supposed to fail as the v10 Astro Cloudflare adapter hasn't been released yet, this PR should be merged after the v10 release (and by that time the e2es should pass)

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included (this flow is tested by the C3 e2es already)
  - [ ] Not necessary because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): <https://github.com/cloudflare/cloudflare-docs/pull/>...
  - [x] Not necessary because: this just makes sure things will work the same when Astro releases their v10 Cloudflare adapter

### CODE REVIEW

**Comment by dario-piotrowicz** [comment]
cc. @alexanderniebuhr 🙂

File: packages/create-cloudflare/templates/astro/c3.ts [hunk 1]
```diff
@@ -46,9 +46,9 @@ const updateAstroConfig = () => {
 			n.node.arguments = [
 				b.objectExpression([
 					b.objectProperty(
-						b.identifier("runtime"),
+						b.identifier("platformProxy"),
 						b.objectExpression([
-							b.objectProperty(b.identifier("mode"), b.stringLiteral("local")),
+							b.objectProperty(b.identifier("enable"), b.booleanLiteral(true)),
```
Comment (alexanderniebuhr) [inline packages/create-cloudflare/templates/astro/c3.ts:51]:
```suggestion
							b.objectProperty(b.identifier("enabled"), b.booleanLiteral(true)),
```

This should be `enabled` and not `enable` (https://github.com/withastro/adapters/pull/159/files#diff-1d2a2b56ad20e7e5b13936929d10bebf30d56ffa7cdc64f686dff4ae6cfafb60R52)

Comment (dario-piotrowicz) [reply packages/create-cloudflare/templates/astro/c3.ts]: 🤦 thanks! 🙂

**Review by IgorMinar — APPROVED** [review]
LGTM. Since this will ship in Astro soon, we should get it out during LDW.

CR: https://jira.cfdata.org/browse/CR-850376

**Comment by petebacondarwin** [comment]
There are quite a few CI failures. I haven't checked but can they all be flakes?

**Comment by alexanderniebuhr** [comment]
FYI, the adapter update was released!

**Comment by dario-piotrowicz** [comment]
> There are quite a few CI failures. I haven't checked but can they all be flakes?

The e2e failures were to be expected since the Astro v10 adapter wasn't release when I opened this PR, the adapter's been release and after re-running the e2e they are all green now 🙂👍

**Comment by IgorMinar** [comment]
I've merged this since CI is green and the PR is already under the LDW exemption

**Pull request merged**